dogm/demo/main.cpp

```diff
@@ -16,31 +16,40 @@
 
 int main(int argc, const char** argv)
 {
-    dogm::GridParams params;
-    params.size = 50.0f;
-    params.resolution = 0.2f;
-    params.particle_count = 3 * static_cast<int>(10e5);
-    params.new_born_particle_count = 3 * static_cast<int>(10e4);
-    params.persistence_prob = 0.99f;
-    params.process_noise_position = 0.1f;
-    params.process_noise_velocity = 1.0f;
-    params.birth_prob = 0.02f;
-    params.velocity_persistent = 30.0f;
-    params.velocity_birth = 30.0f;
+    dogm::GridParams grid_params;
+    grid_params.size = 50.0f;
+    grid_params.resolution = 0.2f;
+    grid_params.particle_count = 3 * static_cast<int>(10e5);
+    grid_params.new_born_particle_count = 3 * static_cast<int>(10e4);
+    grid_params.persistence_prob = 0.99f;
+    grid_params.process_noise_position = 0.1f;
+    grid_params.process_noise_velocity = 1.0f;
+    grid_params.birth_prob = 0.02f;
+    grid_params.velocity_persistent = 30.0f;
+    grid_params.velocity_birth = 30.0f;
 
     dogm::LaserSensorParams laser_params;
     laser_params.fov = 120.0f;
     laser_params.max_range = 50.0f;
     laser_params.resolution = 0.2f;
+    const int sensor_horizontal_scan_points = 100;
+
+    // Simulator parameters
+    const int simulation_steps = 14;
+    const float simulation_step_period = 0.1f;  // TODO use units instead of plain float
```

Comment:
Quick question: What exactly do you mean?

Reply:
Right, I forgot to mention that in the PR: in other projects, I saw that people encode the physical unit in the type, e.g.

```c++
using TimeInS = float;
// ...
const TimeInS simulation_step_period = 0.1f;
```

This is nicely using the otherwise almost meaningless `float` type to denote the variable's unit, and does not need any appendix in the variable name. I would suggest introducing a `units.h` file that contains the using statements, including that where needed, and replacing plain `float` (not all at once, this may e.g. grow organically). What do you think about this?

Reply:
I'm leaning more towards describing the unit a comment, for example in this case a doc for the method `updateParticleFilter`. In my opinion encoding the type is kinda hiding the underlying type. So just having a comment is my prefered style here :)

Reply:
Ok, I understand your reasoning. I removed the comment, let's do it the comment-way in future